### ISSUE TITLE
Real-Time segmentation

### ISSUE THREAD

**Issue by terbed** [WillBrennan/SemanticSegmentation#1]
Hi!

First of all thank you for your work!

I would be interested in real-time skin segmentation!

- Daniel

**Comment by WillBrennan** [comment]
Hey Daniel! 

Its great to see some interest in the project, I've added [BiSeNetV2](https://arxiv.org/pdf/2004.02147.pdf) for real-time segmentation and added a pre-trained skin segmentation model.

It'd be interesting to see what you use it for! 

#2

**Comment by terbed** [comment]
Thank you! I would like to use it for pulse detection from video.

May I ask that on how many samples was the model trained? I will check it now, thanks a lot again!

**Comment by WillBrennan** [comment]
Ah nice! Its around 150 images but with heavy augmentation. If you find any failure cases then feel free to mark them up and send them over, then I can train a new model that fixes those.

**Comment by terbed** [comment]
For my own recordings it is far from accurate. I thought the skin colored background on the left could be a problem, but it generally not really finds the skin. I know these are challenging images. But it couldn't mark my face too.
 
I guess I should train it with large amount of data.

![results_skin_out31](https://user-images.githubusercontent.com/13771341/85692432-acb42d80-b6d5-11ea-94f6-5b0b40abd2f9.png)
![results_skin_out137](https://user-images.githubusercontent.com/13771341/85692438-ade55a80-b6d5-11ea-8444-11f0afc4f454.png)
![results_skin_out140](https://user-images.githubusercontent.com/13771341/85692440-ade55a80-b6d5-11ea-8aa4-ddcb72991c74.png)

**Comment by WillBrennan** [comment]
Hmmm; I'd suggest double checking the performance of the accurate one first. Then play around with the `--threshold` parameter to see if the arms are picked up at a lower threshold. 

If those don't work then feel free to send over the annotated images and I'll add them to the training set. 

The other thing to check is the models dependence on size. If these images are too big or too small you'll see bad performance. It was trained on marked up COCO images so 640x480. I'd suggest resizing your images to that size.

**Comment by WillBrennan** [comment]
![mask_skin_accurate_85692440-ade55a80-b6d5-11ea-8aa4-ddcb72991c74](https://user-images.githubusercontent.com/7763205/85788209-84075480-b724-11ea-81b5-6a99c52c96fe.png)
with a threshold of 0.3 and resizing the image to 40% of the size you'll get a decent segmentation mask with the accurate one. The real-time one performs poorly and needs more data. 
![mask_skin_realtime_85692440-ade55a80-b6d5-11ea-8aa4-ddcb72991c74](https://user-images.githubusercontent.com/7763205/85788270-9d100580-b724-11ea-994d-4b43b42f7da7.png)
